### PR TITLE
[WIP] Add optional pass_key argument to map_tasks

### PR DESCRIPTION
**Pull request by rjzamora**
Adds an optional `pass_key` argument to `map_tasks`.  This enables a simple fix to [distributed#4177](https://github.com/dask/distributed/issues/4177), because it allows the `map_tasks` utility to be used to collect proper task-future dependencies.

- [ ] Tests added / passed
- [ ] Passes `black dask` / `flake8 dask`

### CODE REVIEW

File: dask/blockwise.py [hunk 1]
```diff
@@ -260,11 +260,14 @@ def cull(self, keys, all_hlg_keys):
...
         return self._cached_dict["basic_layer"].cull(keys, all_hlg_keys)
 
-    def map_tasks(self, func):
+    def map_tasks(self, func, pass_key=False):
         new_indices = []
         for key, input_indices in self.indices:
             if input_indices is None:  # A literal
-                new_indices.append((func([key]), None))
+                if pass_key:
+                    new_indices.append((func([key], key), None))
```
Comment (madsbk) [inline dask/blockwise.py:268]: I don't think this is strictly correct. `key` here is a literal and not a graph key. See https://github.com/dask/dask/blob/97c2f060d5ab31d831d20042e0a3e41ede3d2a64/dask/blockwise.py#L380-L382

**Review by madsbk — CHANGES_REQUESTED** [review]
I think this is a great feature @rjzamora but I am a bit concerned with the Blockwise implementation of the key (see my comment).

But if we can get this to work, let's make the `key` argument mandatory. All the `map_tasks()` implementations need to support it anyways.

**Comment by jrbourbeau** [comment]
Is it safe to close this PR since we ended up going another direction in https://github.com/dask/distributed/pull/4178?

**Comment by rjzamora** [comment]
Yeah - It is unlikely we will need this change.

**Pull request closed without merging**